### ISSUE TITLE
`Object::as_tag` does not seems to work as expected. 

### ISSUE THREAD

**Issue by oknozor** [rust-lang/git2-rs#536]
I've been playing with the [tag example](https://github.com/rust-lang/git2-rs/blob/master/examples/tag.rs), the method `Object::as_tag` always return none unless the tag has a message : 
```rust 
if let Some(tag) = obj.as_tag() {
                print_tag(tag, args);
            } else if let Some(commit) = obj.as_commit() {
                print_commit(commit, name, args);
            } else {
                print_name(name);
            }
```
- tagging a commit with `git tag v0.0.0` and trying to convert the corresponding object to a tag will return `None`. 
- tagging a commit with ` git tag -a v0.0.0 -m "some message"` and trying to convert the corresponding object to a tag will return `Some(Tag { name: "v0.0.0", id: oid... })`

Is this an expected behavior ? If it is, It might be a good idea to document it. I'd be happy to submit a PR.

**Comment by ehuss** [comment]
Yes, I believe `ObjectType::Tag` is for "annotated" tags only.  I think "lightweight" tags are just a reference.  PR's to improve the docs are always good!
